mgmtsystem/openstack.py

```diff
@@ -208,21 +210,7 @@ def create_vm(self):
     def delete_vm(self, instance_name):
         self.logger.info(" Deleting OpenStack instance {}".format(instance_name))
         instance = self._find_instance_by_name(instance_name)
-        ip_addr = self.current_ip_address(instance_name)
-        floating_ip = None
-        if ip_addr is not None:
-            floating_ips = self.api.floating_ips.findall(ip=ip_addr)
-            if floating_ips:
-                floating_ip = floating_ips[0]
-                self.logger.info('Detaching floating IP {} from {}'.format(ip_addr, instance_name))
-                instance.remove_floating_ip(floating_ip)
-                wait_for(
-                    lambda: self.current_ip_address(instance_name) is None, delay=1, timeout='1m')
-                self.logger.info('Deleting floating IP {}'.format(ip_addr))
-                floating_ip.delete()
-                wait_for(
-                    lambda: len(self.api.floating_ips.findall(ip=ip_addr)) == 0,
-                    delay=1, timeout='1m')
+        self.delete_floating_ip(instance)
```

Comment:
I think it would probably be better to split this into am assign and a delete operation in two steps. If we want mgmt_system to more universal and people to use it, it makes sense that people may just want to disassociate it, and not kill it. Perhaps we should have a default value for this that is maybe ```delete_fip=True```. This would delete by default, and always disassociate, but give the option to _not_ delete if they so wish.